lib/service/home_widget_service.dart

```diff
@@ -0,0 +1,151 @@
+import 'dart:convert';
+
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/encrypt_env/secrets.g.dart';
+import 'package:autonomy_flutter/model/dailies.dart';
+import 'package:autonomy_flutter/service/feralfile_service.dart';
+import 'package:autonomy_flutter/util/asset_token_ext.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:flutter/services.dart' show ByteData, rootBundle;
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:home_widget/home_widget.dart';
+import 'package:http/http.dart' as http;
+import 'package:nft_collection/graphql/model/get_list_tokens.dart';
+import 'package:nft_collection/models/asset_token.dart';
+import 'package:nft_collection/services/indexer_service.dart';
+
+class HomeWidgetService {
+  final String appGroupId = 'com.bitmark.autonomy_flutter';
+  final String androidWidgetName = 'FeralfileDaily';
+  final String iosWidgetName = 'iosWidgetName'; // TODO: Update this value
+
+  Future<void> init() async {
+    await HomeWidget.setAppGroupId(appGroupId);
+    HomeWidget.widgetClicked.listen((widgetName) {
+      log.info('[HomeWidgetService] Widget clicked: $widgetName');
+    });
+  }
+
+  Future<void> updateWidget(
+      {required Map<String, dynamic> data, bool shouldUpdate = true}) async {
+    data.forEach((key, value) {
+      HomeWidget.saveWidgetData(key, value);
```

Comment:
if you re-save widget data, will it reload the widget, if it heavy, consider `getWidgetData` then compare with the data you are about to save

Reply:
we dont have to. save data does not trigger update widget